### PR TITLE
DB-6679 Merge 2.5.0.1806S branch to branch-2.5

### PR DESCRIPTION
**Pull request by OlegMazurov**
> 2.5.0.1806S was created specifically to fix installation/packaging issues at customer. Now that the initial installation is successful, we need to merge this branch to mainline 2.5. Also, this is going to resolve packaging issues for other customer (which is running HDP2.6.3).

### CODE REVIEW

**Comment by jleach4** [comment]
+1

File: assembly/hdp2.6.3/build/var/lib/ambari-server/resources/common-services/SPLICEMACHINE/2.5.1/service_advisor.py [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+      zoo_cfg = services['configurations']['zoo.cfg']["properties"]
+      print(zoo_cfg),zoo_cfg
+      putZooProperty = self.putProperty(configurations, "zoo.cfg", services)
+      putZooProperty('maxClientCnxns',0)
+      putZooProperty('maxSessionTimeout',120000)
+
+    if 'yarn-site' in services['configurations']:
+      yarn_site = services['configurations']['yarn-site']["properties"]
+      putYarnSiteProperty = self.putProperty(configurations,"yarn-site",services)
+      yarn_application_classpath = yarn_site['yarn.application.classpath'] + ',/var/lib/splicemachine/lib/'
```
Comment (dgomezferro) [inline assembly/hdp2.6.3/build/var/lib/ambari-server/resources/common-services/SPLICEMACHINE/2.5.1/service_advisor.py]: We wanted to remove splice from yarn's application classpath, right?

File: assembly/hdp2.6.3/build/var/lib/ambari-server/resources/common-services/SPLICEMACHINE/2.5.1/service_advisor.py [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+        "hbase.splitlog.manager.timeout" : "3000",
+        "hbase.status.multicast.port" : "16100",
+        "hbase.wal.disruptor.batch" : "true",
+        "hbase.wal.provider" : "multiwal",
+        "hbase.wal.regiongrouping.numgroups" : "16",
+        "hbase.zookeeper.property.tickTime" : "6000",
+        "hfile.block.bloom.cacheonwrite" : "TRUE",
+        "io.storefile.bloom.error.rate" : "0.005",
+        "splice.authentication.native.algorithm" : "SHA-512",
+        "splice.authentication" : "NATIVE",
```
Comment (dgomezferro) [inline assembly/hdp2.6.3/build/var/lib/ambari-server/resources/common-services/SPLICEMACHINE/2.5.1/service_advisor.py]: Does the client get a warning if he changes any of these properties?

File: assembly/mapr6.0.0/src/main/resources/conf/hbase-site.xml.patch [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
++  <property><name>hbase.regions.slop</name><value>0.01</value></property>
++  <property><name>hbase.regionserver.codecs</name><value></value></property>
++  <property><name>hbase.regionserver.global.memstore.lowerLimit</name><value>0.38</value></property>
++  <property><name>hbase.regionserver.global.memstore.size.lower.limit</name><value>0.9</value></property>
++  <property><name>hbase.regionserver.global.memstore.size</name><value>0.25</value></property>
++  <property><name>hbase.regionserver.global.memstore.upperLimit</name><value>0.4</value></property>
++  <property><name>hbase.regionserver.hlog.blocksize</name><value>134217728</value></property>
++  <property><name>hbase.regionserver.logroll.period</name><value>3600000</value></property>
++  <property><name>hbase.regionserver.maxlogs</name><value>48</value></property>
++  <property><name>hbase.regionserver.metahandler.count</name><value>10</value></property>
```
Comment (dgomezferro) [inline assembly/mapr6.0.0/src/main/resources/conf/hbase-site.xml.patch]: metahandler has to be higher

File: assembly/hdp2.6.3/build/var/lib/ambari-server/resources/common-services/SPLICEMACHINE/2.5.1/service_advisor.py [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+
+    }
+    return hdfs_site_desired_values
+
+  def getHBaseSiteDesiredValues(self):
+    hbase_site_desired_values = {
+        "hbase.coprocessor.master.classes" : "org.apache.ranger.authorization.hbase.RangerAuthorizationCoprocessor,com.splicemachine.hbase.SpliceMasterObserver",
+        "hbase.regionserver.global.memstore.size" : "0.25",
+        "hfile.block.cache.size" : "0.25",
+        "hbase.regionserver.handler.count" : "200",
```
Comment (dgomezferro) [inline assembly/hdp2.6.3/build/var/lib/ambari-server/resources/common-services/SPLICEMACHINE/2.5.1/service_advisor.py]: check metahandlers too

**Comment by stpabhi** [comment]
@jleach4 @dgomezferro Can we have CDH5.14 support as part of this? #1485

**Comment by jleach4** [comment]
@stpabhi I am very unpopular with this monster pull request as is.  How about we wait until this is merged and then we can add 5.14?  Is that OK?

**Comment by stpabhi** [comment]
@jleach4 Thanks, it can wait :)

File: assembly/hdp2.6.3/build/var/lib/ambari-server/resources/common-services/SPLICEMACHINE/2.5.1/metainfo.xml [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+                <config-type>zeppelin-env</config-type>
+                <config-type>zoo.cfg</config-type>
+                <config-type>yarn-site</config-type>
+            </configuration-dependencies>
+
+            <requiredServices>
+                <service>HBASE</service>
+                <service>HDFS</service>
+                <service>YARN</service>
+                <service>ZOOKEEPER</service>
```
[the remark this answers is not in the document]
Comment (jleach4) [reply assembly/hdp2.6.3/build/var/lib/ambari-server/resources/common-services/SPLICEMACHINE/2.5.1/metainfo.xml]: let me fix some of these items, one sec.

File: assembly/hdp2.6.3/build/var/lib/ambari-server/resources/common-services/SPLICEMACHINE/2.5.1/service_advisor.py (hunk 1 above)
Comment (jleach4) [reply assembly/hdp2.6.3/build/var/lib/ambari-server/resources/common-services/SPLICEMACHINE/2.5.1/service_advisor.py]: That is correct.

File: assembly/hdp2.6.3/build/var/lib/ambari-server/resources/common-services/SPLICEMACHINE/2.5.1/service_advisor.py (hunk 2 above)
Comment (jleach4) [reply assembly/hdp2.6.3/build/var/lib/ambari-server/resources/common-services/SPLICEMACHINE/2.5.1/service_advisor.py]: yes he does if it is different than ours.

File: assembly/mapr6.0.0/src/main/resources/conf/hbase-site.xml.patch (hunk 3 above)
Comment (jleach4) [reply assembly/mapr6.0.0/src/main/resources/conf/hbase-site.xml.patch]: setting to 200 per doc.

**Comment by jleach4** [comment]
@dgomezferro Made the changes per your recommendations

**Comment by jyuanca** [comment]
@OlegMazurov Please resolve conflicts.

**Pull request merged**